compiler-rt/lib/asan/asan_poisoning.cpp

```diff
@@ -20,11 +20,33 @@
 #include "sanitizer_common/sanitizer_flags.h"
 #include "sanitizer_common/sanitizer_interface_internal.h"
 #include "sanitizer_common/sanitizer_libc.h"
+#include "sanitizer_common/sanitizer_stackdepot.h"
 
 namespace __asan {
 
 static atomic_uint8_t can_poison_memory;
 
+static Mutex PoisonRecordsMutex;
+static PoisonRecordRingBuffer *PoisonRecords = nullptr;
+
+void InitializePoisonTracking() {
+  if (flags()->track_poison <= 0)
+    return;
+
+  PoisonRecords = PoisonRecordRingBuffer::New(flags()->track_poison);
+}
+
+PoisonRecordRingBuffer *SANITIZER_ACQUIRE(PoisonRecordsMutex)
```

Comment:
please check __asan::InstallAtForkHandler

Probably we need `Lock/Unlock PoisonRecords();` there

Reply:
Done

Reply:
Fix "No need to AcquirePoisonRecords" and "if (fork_child)" as discussed.

Reply:
> Fix "No need to AcquirePoisonRecords" and "if (fork_child)" as discussed.

Done